### PR TITLE
Rename "Closed Jobs" to "Previous Applications"

### PR DESCRIPTION
**Pull request by jaspax**
**Release Type**: UI Improvement

Fixes https://x-team-internal.atlassian.net/browse/XP-3474

## Description

* Change the user-visible wording "Closed Applications" to "Previous Applications".
* Update component names from `ClosedJobApplications` to `PreviousJobApplications`
* Rename component files to align with component names
* Update other relevant variable names from `closed` to `previous`

## Checklist

- [x] set yourself as an assignee
- [x] set appropriate labels for a PR (`In Review` or `In Progress` depending on its status)
- [x] move respective JIRA issue to the `PEER REVIEW` column
- [x] make sure your code lints (`npm run lint`)
- [x] Flow typechecks passed (`npm run flow`)
- [x] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases and/or debug cases
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot (**NOTE:** This is temporarily disabled. Skip it.)
- [x] once finished with all the changes, you run `npm run build` command and committed changed files to the branch

## Steps to Test or Reproduce

Start the storybook app in XP-UI, then visit http://localhost:9001/?path=/story/screens-and-layouts-jobspagelayout-use-cases--my-applications-page. Scroll down to the bottom to check out the new wording.

## Impacted Areas in Application

JobsPageLayout, the "My Applications" area

### CODE REVIEW

**Comment by nicksp** [comment]
@jaspax I can see a bunch of conflicts. Could you please resolve them? I believe you'll need to rebase with `develop` and re-run `npm run build` command.

File: src/components/ui/PreviousJobApplications.stories.js [hunk 1]
```diff
@@ -5,9 +5,9 @@ import { storiesOf } from '@storybook/react'
 import { object, text } from '@storybook/addon-knobs'
 import faker from 'faker'
 
-import ClosedJobApplications from './ClosedJobApplications'
+import PreviousJobApplications from './PreviousJobApplications'
 
-export const closedJobApplicationsJobsSample = [
+export const PreviousJobApplicationsJobsSample = [
```
Comment (nicksp) [inline src/components/ui/PreviousJobApplications.stories.js]: Could we please keep names consistent, ie. `previousJobApplicationsJobsSample` (small `p` in the beginning). It's a regular identifier which usually starts with a small letter unless there's a strong reason to do differently.

File: src/components/ui/PreviousJobApplications.stories.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const closedJobApplicationsJobsSample = [
   }
 ]
 
-export const closedJobApplicationsMessageSample = 'Applications older than 3 months are archived'
+export const PreviousJobApplicationsMessageSample = 'Previous applications older than 3 months are archived'
```
Comment (nicksp) [inline src/components/ui/PreviousJobApplications.stories.js]: Let's keep the first letter small. Only classes and components start with a capital first letter.

Comment (jaspax) [reply src/components/ui/PreviousJobApplications.stories.js]: Good catch, that's a result of doing search-and-replace, and I didn't notice it during my manual code review.

**Pull request merged**